app/graphql/types/doi_item.rb

```diff
@@ -460,9 +460,9 @@ def contributors(**args)
         "given_name" => c.fetch("givenName", nil),
         "family_name" => c.fetch("familyName", nil),
         "affiliation" =>
-          c.fetch("affiliation", []).map do |a|
+          Array.wrap(c.fetch("affiliation", [])).map do |a|
             { "id" => a["affiliationIdentifier"], "name" => a["name"] }.compact
-          end,
+          end.compact,
```

Comment:
Will this `.compact` fire properly in all the situations where there might be a `null` value in the array?

If for some reason there is a `null` in the wrapped array, this looks like it would currently pass through to the `map` function first, which might lead to an error, rather than being removed by `.compact`.

I think the way to prevent that is to move (or add another) `.compact` between the `Array.wrap()` and `map`, but the big caveat here is that I am still learning about the fun intricacies of Ruby in this regard, so it would probably be worth writing a spec test first to see if it would indeed fail under the circumstances above before making any changes,

Reply:
I imagine actually it'll be fine. It's a little odd but I imagine
[nil] maps to { "id" => nil, "name" => nil } which compact sto just [nil] again.
So the last compact then makes it []

The other way to solve this would be to just compact before the map I think? I'd have to actually run the code but I'd have thought if you compact before doing map then it wouldnt even go into map as it'd be nil.

Although with what Cody has done this does potentially catch scenario where you have a hash which has nil values for it's parts.
so:
[{ "affiliationidentifier" => nil, "name" => nil}] maps to { "id" => nil, "name" => nil } which compacts to nil again.

So in summary maybe go with this, for extra care like Mike said you could write a spec to handle these scenarios.

Reply:
For context, the general strategy here was to mimic the serialization on the REST side: https://github.com/datacite/lupo/blob/344ccdecd1998518411084d75140c56e78ea3a6d/app/serializers/datacite_doi_serializer.rb#L127 

Testing for nil values within an array in the affiliation property is a little challenging because I don't think the XML generation code can handle the nil values. I get this error when trying:

```     NoMethodError:
       undefined method `[]' for nil:NilClass
     # ./app/models/concerns/crosscitable.rb:134:in `update_xml'
     # ./spec/graphql/types/work_type_spec.rb:897:in `block (3 levels) in <top (required)>'```

I'm sure there's a workaround — can explore more if that makes sense.